### PR TITLE
QtWebEngine: Fix crash-on-exit.

### PR DESCRIPTION
**Pull request by JuBan1**
There was a SIGSEV that popped up in the experimental branch. I fixed it and broke editor signaling with it. When I fixed editor signaling I made the SIGSEV come back.

So here's the fix for the fix for the fix for the SIGSEV.

I should go into gardening, not programming.

------

Technical details:

I'm not quite sure why this happens but here's what I believe: When `TopEditorContainer` is destroyed, it first destructs its members and then calls its parent `QObject` destructor. That one makes sure all children get destroyed as well, that includes all `EditorTabWidgets` and the tabs therein. During that process, a number of signals are emitted that are being caught and acted upon by the now half-destructed `TopEditorContainer`. That kills the program.

Manually disconnecting the affected signals at destruction leads to a SIGPIPE instead (first time I ever managed that error!), probably because of some meta-type magic done by Qt. So instead, I added a switch to stop handling the problematic signal when the `TopEditorContainer` is being destructed.

I think it'll suffice for now, the whole tab stuff is due for an overhaul anyways.

### CODE REVIEW

**Comment by Teklad** [comment]
YOU BROKE TRAVIS!  Lol... kidding, but this will do fine until we can get QWebEngine integration complete...  The next big project is Editor/Tab overhaul... I'm just @danieleds will be falling out of his chair with that PR. xD

**Comment by JuBan1** [comment]
At least this doesn't seem to be my fault. Way to be an ass, Travis.

File: src/ui/include/topeditorcontainer.h [hunk 1]
```diff
@@ -49,6 +51,7 @@ class TopEditorContainer : public QSplitter
 private:
     EditorTabWidget *m_currentTabWidget;
     int m_currentTabIndex = -1;
+    bool m_shuttingDown = false;
```
Comment (danieleds) [inline src/ui/include/topeditorcontainer.h]: Add a big comment summing up your "Technical details" paragraph?

Comment (JuBan1) [inline src/ui/include/topeditorcontainer.h]: Done

**Pull request merged**